plasma_framework/contracts/src/exits/payment/controllers/PaymentProcessInFlightExit.sol

```diff
@@ -259,6 +247,54 @@ library PaymentProcessInFlightExit {
         framework.batchFlagOutputsFinalized(outputIdsToFlag);
     }
 
+    function returnInputPiggybackBonds(
+        Controller memory self,
+        PaymentExitDataModel.InFlightExit storage exit,
+        address token
+    )
+        private
+    {
+        for (uint16 i = 0; i < PaymentTransactionModel.MAX_INPUT_NUM(); i++) {
+            PaymentExitDataModel.WithdrawData memory withdrawal = exit.inputs[i];
+
+            // If the input has been challenged, isInputPiggybacked() will return false
+            if (token == withdrawal.token && exit.isInputPiggybacked(i)) {
+                bool success = SafeEthTransfer.transferReturnResult(
+                    withdrawal.exitTarget, withdrawal.piggybackBondSize, self.safeGasStipend
+                );
+
+                // we do not want to block a queue if bond return is unsuccessful
+                if (!success) {
+                    emit InFlightBondReturnFailed(withdrawal.exitTarget, withdrawal.piggybackBondSize);
+                }
+            }
+        }
+    }
+
+    function returnOutputPiggybackBonds(
+        Controller memory self,
+        PaymentExitDataModel.InFlightExit storage exit,
+        address token
+    )
+        private
+    {
+        for (uint16 i = 0; i < PaymentTransactionModel.MAX_OUTPUT_NUM(); i++) {
```

Comment:
Why looping over `PaymentTransactionModel.MAX_OUTPUT_NUM` and not the length of outputs?

Reply:
Agree. With RLP decoding fixed - lets iterate through length of inputs.

Reply:
So `exit.inputs.length` and `exit.outputs.length` would be exactly same as `MAX_INPUT_NUM` and `MAX_OUTPUT_NUM` as that is how it is defined in the storage:
https://github.com/omisego/plasma-contracts/blob/bfbf7cc76cb2ba4564cc1e922dbc3a9bacb8c725/plasma_framework/contracts/src/exits/payment/PaymentExitDataModel.sol#L73-L74

However, agree it is more self explaining if using `in/outputs.length` instead.